subprojects/docs/src/docs/userguide/migrating_from_maven.adoc

```diff
@@ -0,0 +1,588 @@
++
+Dependencies required for test compilation should be declared against the `testImplementation` configuration.
+Those that are only required for running the tests should use `testRuntimeOnly`.
+
+`provided`::
+Use the `compileOnly` configuration.
++
+Note that the <<war_plugin#sec:war_dependency_management,War Plugin>> adds `providedCompile` and `providedRuntime` dependency configurations.
+These behave slightly differently from `compileOnly` and simply ensure that those dependencies aren't packaged in the WAR file.
+However, the dependencies are included on runtime and test runtime classpaths, so use these configurations if that's the behavior you need.
+
+`import`::
+The `import` scope can be used within both `<dependencyManagement>` and `<dependencies>` blocks.
+If you're dealing with the former scenario, read the section on <<migmvn:using_boms,Using bills of materials>>.
++
+The latter scenario has no equivalent in Gradle.
+You can specify a regular dependency on the POM or its library, but the dependencies declared in the `<dependencies>` block will be treated as normal transitive dependencies of the build.
+In other words, those dependencies in the POM will be mapped to the appropriate classpath based on the declared configuration.
++
+For example, imagine you want to use the `groovy-all` POM for your tests.
+It's solely a POM that has its own dependencies listed inside a `<dependencies>` block.
+The appropriate configuration in the Gradle build looks like this:
++
+.Consuming a POM dependency that has only transitive dependencies
+====
+include::sample[dir="userguide/mavenMigration/basic/groovy",files="build.gradle[tags=pom-dependencies]"]
+include::sample[dir="userguide/mavenMigration/basic/kotlin",files="build.gradle.kts[tags=pom-dependencies]"]
+====
++
+The result of this will be that all `compile` and `runtime` scope dependencies in the `groovy-all` POM get added to the test runtime classpath, while only the `compile` scope dependencies get added to the test compilation classpath.
+Dependencies with other scopes will be ignored.
+
+[[migmvn:declaring_repos]]
+=== Declaring repositories
+
+Gradle allows you to retrieve declared dependencies from any Maven-compatible or Ivy-compatible repository.
+Unlike Maven, it has no default repository, so you have to declare at least one.
+In order to have the same behavior as your Maven build, just configure <<repository_types#sub:maven_central,Maven Central>> in your Gradle build, like this:
+
+.Configuring the build to use Maven Central
+====
+include::sample[dir="userguide/mavenMigration/basic/groovy",files="build.gradle[tags=repositories]"]
+include::sample[dir="userguide/mavenMigration/basic/kotlin",files="build.gradle.kts[tags=repositories]"]
+====
+
+You can also use the `repositories {}` block to configure custom repositories, as described in the <<repository_types#sub:maven_repo,Repository Types>> chapter.
```

Comment:
Maybe add a section to explain the differences between Gradle dependency cache and Maven's `.m2`, which explain why it's usually a bad idea to add `mavenLocal()` as a repository. Key differences are:

1. Gradle dependency cache is aware of where a dependency was sourced from (what is its original source)
2. `.m2` can be populated by dependencies declared in a different project, which creates unreliable builds. The dependency cache doesn't have such an issue because of 1.
3. the Gradle dependency cache is not relocatable/sharable
4. Gradle doesn't use the dependency cache as a way to consume inter-project dependencies. You never do `install`, it's redundant.

Reply:
Speaking of which, I should clarify in the _Understanding the build lifecycle_ section that `install` is redundant from the perspective of Gradle builds and should only be used to interoperate with Maven builds. Does that sound right?

Reply:
Not mentioning that `install` in Gradle is _not_ the equivalent of `install` in Maven.

Reply:
Is there a standard `install` task in Gradle?